### PR TITLE
Dynamically generate nonces

### PR DESCRIPTION
**Pull request by wolfejw86**
Possible implementation that addresses #89

I will not be overly sad if this is simply not something this package wants to do, and instead just encourage people to use the base `helmet` package by itself.  Feel free to close this and disregard if that is the case :)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,11 +2,35 @@
...
+
+      options.contentSecurityPolicy.directives.scriptSrc.push(`'${nonce}'`)
+      options.contentSecurityPolicy.directives.styleSrc.push(`'${nonce}'`)
+
+      if (!reply.raw.locals) {
+        reply.raw.locals = {}
+      }
+
+      // TODO - what is the best place to put this?
+      reply.raw.locals.nonce = nonce
```
Comment (wolfejw86) [inline index.js:29]: This is actually a deeper question for me - I still haven't learned what the "fastify" equivalent way of doing what I used to do in express with `res.locals`
What is the best place to put these nonces? (and a general question from me - where would you generally store "middleware" mutation type data?)

Comment (jsumners) [reply index.js:29]: What is `res.locals`?

**Comment by jsumners** [comment]
Given the comment in the original issue, I'm going to hold off reviewing this until a decision is made upstream regarding adding the feature back to `helmet`.

**Comment by wolfejw86** [comment]
Yes I would also agree - it seems like this will just get closed since the helmet package will re-add the missing functionality

**Comment by Eomm** [comment]
It seems that helmet released it in 4.1

We could just add some test and update the depth as improvement and supported feature to check in next major version

**Comment by fox1t** [comment]
@wolfejw86 to resolve that conflict in types you can write them like this:
```ts
import { FastifyPlugin } from "fastify";
import helmet from "helmet";

interface FastifyHelmetOptions extends NonNullable<Parameters<typeof helmet>[0]> {
  generateNonces?: boolean
}

export const fastifyHelmet: FastifyPlugin<FastifyHelmetOptions>;

export default fastifyHelmet;
```

**Comment by mcollina** [comment]
I think this can be closed as this functionality landed back inside helmet.

**Comment by wolfejw86** [comment]
Agree 100% - glad it’s not something fastify-helmet will have to maintain

**Pull request closed without merging**